codebundles/k8s-podresources-health/runbook.robot

```diff
@@ -0,0 +1,106 @@
+    [Documentation]    Scans a list of pods in a namespace using labels as a selector and checks if their resources are set.
+    [Tags]    Pods    Resources    Resource    Allocation    CPU    Memory    Startup    Initialization    Prehook    Liveness    Readiness
+    ${pods_without_resources}=    RW.CLI.Run Cli
+    ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} get pods --context=${CONTEXT} -n ${NAMESPACE} ${LABELS} --field-selector=status.phase=Running -ojson | jq '[.items[] | select(.spec.containers[].resources == {} or (.spec.containers[].resources|has("requests")|not) or (.spec.containers[].resources|has("limits")|not)) | {namespace:.metadata.namespace,name:.metadata.name}]'
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${kubeconfig}
+    ...    target_service=${kubectl}
+    ...    render_in_commandlist=true
+    ${no_resource_count}=    RW.CLI.Parse Cli Json Output
+    ...    rsp=${pods_without_resources}
+    ...    extract_path_to_var__no_resources_count=length(@)
+    ...    set_issue_title=Found pod without resources specified!
+    ...    set_severity_level=4
+    ...    no_resources_count__raise_issue_if_gt=0
+    ...    assign_stdout_from_var=no_resources_count
+    ${history}=    RW.CLI.Pop Shell History
+    ${pod_count}=    Convert To Number    ${no_resource_count.stdout}
+    ${summary}=    Set Variable    No pods with unset resources found!
+    IF    ${pod_count} > 0
+        ${summary}=    Set Variable    ${pod_count} pods found without resources specified:\n${pods_without_resources.stdout}        
+    END
+    RW.Core.Add Pre To Report    ${summary}
+    RW.Core.Add Pre To Report    Commands Used:\n${history}
+
+Get Labeled Container Top Info
+    [Documentation]    Performs and a top command on list of labeled workloads to check pod resources.
+    [Tags]    Top    Resources    Utilization    Pods    Workloads    CPU    Memory    Allocation    Labeled
+    ${labeled_pods}=    RW.CLI.Run Cli
+    ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} get pods ${LABELS} -n ${NAMESPACE} --context ${CONTEXT} -o=name --field-selector=status.phase=Running
+    ...    target_service=${kubectl}
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${KUBECONFIG}
+    ...    render_in_commandlist=true
+    ${labeled_pods}=    Split String    ${labeled_pods.stdout}
+    ${labeled_pods}=    Evaluate    [full_name.split("/")[-1] for full_name in ${labeled_pods}]
+    ${resource_util_info}=    Set Variable    No resource utilization information could be found!
+    IF    len(${labeled_pods}) > 0
+        ${temp_top}=    RW.CLI.Run Cli
+        ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} top pod {item} -n ${NAMESPACE} --context ${CONTEXT} --containers
+        ...    target_service=${kubectl}
+        ...    env=${env}
+        ...    secret_file__kubeconfig=${KUBECONFIG}
+        ...    loop_with_items=${labeled_pods}
+        ${resource_util_info}=    Set Variable    ${temp_top.stdout}
+    END
```

Comment:
If this is to be useful & rendered in command list/assist/thing we haven't named properly - then I'd suggest the for loop pattern; something like this (by example) would likely produce the same output; 

`for pod in $(kubectl get pods -n awx -o  custom-columns=":metadata.name" --field-selector=status.phase=Running); do kubectl top pod $pod -n awx --containers; done`

For this code, likely this;
```
for pod in $(${KUBERNETES_DISTRIBUTION_BINARY} get pods ${LABELS} -n ${NAMESPACE} --context ${CONTEXT} -o custom-columns=":metadata.name" --field-selector=status.phase=Running); do ${KUBERNETES_DISTRIBUTION_BINARY} top pod $pod -n ${NAMESPACE} --context ${CONTEXT} --containers; done
```